iceoryx_posh/test/moduletests/test_posh_runtime_node.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/runtime/node.hpp"
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using namespace iox::roudi;
+
+
+namespace iox
+{
+namespace test
+{
+/// @brief Test goal: This test suit verifies class node
+
+class PoshRuntimeNode_test : public Test
+{
+  public:
+    PoshRuntimeNode_test()
+    {
+    }
+
+    virtual ~PoshRuntimeNode_test()
+    {
+    }
+
+    virtual void SetUp()
+    {
+    };
+
+    virtual void TearDown()
+    {
+    };
+
+    const ProcessName_t m_runtimeName{"App"};
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_runtime{&iox::runtime::PoshRuntime::initRuntime(m_runtimeName)};
+};
+
+TEST_F(PoshRuntimeNode_test, ConstructorNodeIsSuccess)
```

Comment:
You are testing here not that the node can be constructed successfully, you test that a constructed node has the correct node name which you verify with `getNodeName`.
Here you also have to think of.
- Zero (empty node name)
- One/Many (makes sense to put them together and you already tested it)
- CornerCases (does a node name allow every possible character, if not we need to test this)
- Limits (test it with a node name of maximum allowed length)

But here I think it is possible that @FerdinandSpitzschnueffler already has taken care of this.

Reply:
Empty node name doesn't have any impact
Every possible character is allowed
Max length cannot be tested because there is a build error.

Reply:
@elBoberido didn't you do some compile time tests?

Reply:
@marthtz I've created an issue, but didn't implemented it yet